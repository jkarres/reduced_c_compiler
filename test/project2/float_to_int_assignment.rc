function : float getfloat() { return 3; }

function : void print(float f) { cout << f << endl; }

const int i = 10;

function : void main() {

  int j = 11;

  //
  // initialization
  //

  float a = 4;
  cout << a << endl;

  float b = i;
  cout << b << endl;

  float c = j;
  cout << c << endl;

  const float d = 4;
  cout << d << endl;

  const float e = i;
  cout << e << endl;

  //
  // assignment
  //

  a = 5;
  cout << a << endl;

  a = i;
  cout << a << endl;

  a = j;
  cout << a << endl;

  //
  // function return value
  //

  cout << getfloat() << endl;

  //
  // passing by value
  //
  print(25);

}
