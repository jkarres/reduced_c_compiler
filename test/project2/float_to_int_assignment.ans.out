4.00
10.00
11.00
4.00
10.00
5.00
10.00
11.00
3.00
25.00
